quic/transport/ngtcp2/connection/openstate.nim

```diff
@@ -31,13 +31,17 @@ proc openServerConnection*(local, remote: TransportAddress,
 
 method enter(state: OpenConnection, connection: QuicConnection) =
   procCall enter(ConnectionState(state), connection)
-  state.quicConnection = some connection
-  state.ngtcp2Connection.onNewId = some proc(id: ConnectionId) =
+  state.quicConnection = Opt.some(connection)
+  # Workaround weird bug
+  proc onNewId(id: ConnectionId) =
     if isNil(connection.onNewId): return
     connection.onNewId(id)
-  state.ngtcp2Connection.onRemoveId = some proc(id: ConnectionId) =
+  state.ngtcp2Connection.onNewId = Opt.some(onNewId)
```

Comment:
fyi @arnetheduck, I was getting compiler crashes when I tried to do it like it was with std/options, but I couldn't get a minimal repro

Reply:
ok - hm, strange construct, using `nil` checks and `Opt` at the same time but ok, shouldn't crash - got a failing branch?

Reply:
https://github.com/status-im/nim-quic/commit/b7488f7ae873cda497acd707cd4d776262363d7d

With 1.2:
```
/home/tavon/Status/nim-quic/quic/transport/ngtcp2/connection/openstate.nim(38, 25) Error: internal error: expr: param not init id_47190008
No stack traceback available
To create a stacktrace, rerun compilation with ./koch temp c <file>
```

With devel:
```
/home/tavon/Status/nim-quic/quic/transport/ngtcp2/connection/openstate.nim(40, 48) Error: redefinition of ':anonymous'; previous declaration here: /home/tavon/Status/nim-quic/quic/transport/ngtcp2/connection/openstate.nim(36, 45)
```

Sounds like a nim bug in the first place (it's because there are two similar procedures in the same scope, I think)

But it didn't happened with std `some`, so maybe `Option.some` isolate the code better than `Opt.some`?

Reply:
`Opt.some` is generic / typebased making it explore the more fragile parts of the compiler

Reply:
`proc some*[T](val: T): Option[T] {.inline.} =`
`template some*[T](O: type Opt, v: T): Opt[T] =`

Both are generic, but Opt is a template too, maximizing the change for failure :)
Switching to `proc some*[T](O: type Opt, v: T): Opt[T] {.inline.} =` fixes this issue, not sure it's worth it